lib/fetchContent.js

```diff
@@ -323,11 +325,137 @@ function getRecentArticles(days, aspects, facets)
 		});
 }
 
+function getArticleTopics(days, facet, aspects, facets)
+{
+	const date = time.getDatetimeRange('days', days, 0);
+	const params = {
+		queryString: `lastPublishDateTime:> ${date.next}`,
+		maxResults : 100,
+		aspects : aspects,
+		facets : {
+			"names": facets,
+			"maxElements":-1
+		}
+	};
+
+	return searchDeeper(params, 10)
+		.then(results =>  extractArticles(results) )
+		.then(articles => sortArticlesIntoFacets(articles, facet) )
+		.then(topicArticle => wrapResults(topicArticle, days, facet) )
+		.catch(err => {
+			throw err;
+		});
+}
+
+function wrapResults(results, days, facet){
+	return {
+		description: `List of topics, from the last ${days} days. Each topic has a list of articles it features in and a list of other topics listed by those articles`,
+		facet: facet,
+		facetItemTotal: results.facetItemCount,
+		articleTotal: results.articleCount,
+		breakdown: results.breakdown
+	};
+}
+
+
+function sortArticlesIntoFacets(articles, facetName = 'topics')
+{
+	let facetWithArticles = [];
+	let totalArticleCount = 0;
+
+	articles.forEach(article => {
+		let facets = article.metadata[facetName];
+
+		if(!!facets){
+			facets.forEach(facet => {
+
+				let itemObj = objs.getItem(facetWithArticles, 'facetName', facet.term.name);
+
+				if(itemObj === undefined){
+					facetWithArticles.push( newCollection(facet.term.name) );
+					itemObj = objs.getItem(facetWithArticles, 'facetName', facet.term.name);
+				}
+
+				itemObj.articleCount = itemObj.articleCount++;
+				itemObj.articles.push(article);
+
+
+				// related topics
+				let articlesTopics = article.metadata.topics;
+				if(articlesTopics){
+					articlesTopics.forEach(topic => {
+						if(topic.term.name != itemObj.facetName){
+							itemObj.relatedTopics.push(topic.term.name);
+						}
+					});
+				}
+
+				//related people
+				let articlesPeople = article.metadata.people;
+				if(articlesPeople){
+					articlesPeople.forEach(person => {
+						if(person.term.name != itemObj.facetName){
+							itemObj.relatedPeople.push(person.term.name);
+						}
+					});
+				}
+
+				//related orgs
+				let articlesOrgs = article.metadata.organisations;
+				if(articlesOrgs){
+					articlesOrgs.forEach(org => {
+						if(org.term.name != itemObj.facetName){
+							itemObj.relatedOrgs.push(org.term.name);
+						}
+					});
+				}
+
+			});
+		}
+
+		totalArticleCount++;
+	});
+
+
+	//count facets
+	facetWithArticles.forEach(item => {
+		item.relatedTopicCount = arrs.uniqueSort(item.relatedTopics, "count");
+		item.relatedPeopleCount = arrs.uniqueSort(item.relatedPeople, "count");
+		item.relatedOrgsCount = arrs.uniqueSort(item.relatedOrgs, "count");
+		item.articleCount = item.articles.length;
+		delete item.relatedTopics;
```

Comment:
if you organised the object as `item.related.relatedTopics`, `item.related.relatedPeople` and `item.related.relatedOrgs` you could do a single delete on `item.related`

Reply:
Ah yes, amended that